packages/data-stores/src/wpcom-plans-ui/selectors.ts

```diff
@@ -1,3 +1,7 @@
 import type { State } from './reducer';
 
 export const isDomainUpsellDialogShown = ( state: State ) => !! state.showDomainUpsellDialog;
+export const getStorageAddOnForPlan = ( state: State ) => ( plan: string ) => {
```

Comment:
Just out of curiosity because I haven't seen this before, why do we need a function returning a function? Can't we supply the `plan` to `getStorageAddOnForPlan` directly?

Reply:
> Can't we supply the plan to getStorageAddOnForPlan directly?

Thank you @Sharpirate . Good catch, and yes, we should. No need for currying here.

@jeyip

Reply:
Thanks y'all! Resolved in https://github.com/Automattic/wp-calypso/pull/80158/commits/072842751fed173b43d931f19fa69fb849deb274